utils/recommendation/widedeep.py

```diff
@@ -0,0 +1,129 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) 2022, Ampere Computing LLC
+
+import math
+import time
+import pickle
+import bz2
+import collections
+
+import numpy as np
+import tensorflow as tf
+
+import utils.misc as utils
+
+
+class WideDeep:
+    """
+    A class providing facilities for preprocessing and postprocessing of ImageNet validation dataset.
+    """
+
+    def __init__(self, batch_size: int, config=None, runner=None, dataset_path=None, tfrecords_path=None):
+
+        if tfrecords_path is None:
+            env_var = "WIDEDEEP_TFRECORDS_PATH"
+            tfrecords_path = utils.get_env_variable(
+                env_var, f"Path to tfrecords path has not been specified with {env_var} flag")
+
+        self.batch_size = batch_size
+        if self.batch_size in [1, 2, 4, 8, 16, 32, 50, 64, 100, 128, 200, 256]:
+            if dataset_path is None:
+                env_var = "WIDEDEEP_DATASET_PATH"
+                dataset_path = utils.get_env_variable(
+                    env_var, f"Path to widedeep dataset has not been specified with {env_var} flag")
+
+        self.tfrecords_path = tfrecords_path
+        self.dataset_path = dataset_path
+        self.current_feature = 0
+        self.correct = 0
+        self.available_instances = sum(1 for _ in tf.compat.v1.python_io.tf_record_iterator(
+            self.tfrecords_path))
+
+        if self.batch_size in [1, 2, 4, 8, 16, 32, 50, 64, 100, 128, 200, 256]:
```

Comment:
you already did this check at this stage - what's crucial for unpickle() to work directly is rather whether the dataset_path is not None